lib/helpers/indexes/filterIndexes.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+function filterOutNonDiscriminatorSchemaIndexes(model, schemaIndexes) {
+  return filterOutNonDiscriminatorIndexes(model, schemaIndexes, { isSchemaIndexes: true });
+}
+
+function filterOutNonDiscriminatorDBIndexes(model, dbIndexes) {
+  return filterOutNonDiscriminatorIndexes(model, dbIndexes, { isSchemaIndexes: false });
+}
+
+module.exports = {
+  filterOutNonDiscriminatorSchemaIndexes,
+  filterOutNonDiscriminatorDBIndexes
+};
+
+function filterOutNonDiscriminatorIndexes(model, indexes, { isSchemaIndexes }) {
+  if (!model.baseModelName) {
+    return indexes;
+  }
+  const discriminatorNameKey = model.schema.discriminatorMapping.key;
+  const discriminatorNameValue = model.schema.discriminatorMapping.value;
+
+
+  return indexes.filter(index => {
```

Comment:
nit:  Isnt it possible to integrate the functionality of getPartialFilterExpress into the filter/ or put the whole logic of that fn into the getPartialFilterExpress? Seems to me that the separation of the functionality is actually artificial.